cartesian_trajectory_controller/include/cartesian_trajectory_controller/control_policies.hpp

```diff
@@ -261,7 +261,27 @@ namespace cartesian_ros_control
       return false;
     };
 
-    ik_solver_ = std::make_unique<KDL::ChainIkSolverPos_LMA>(robot_chain_);
+    // Load user specified inverse kinematics solver
+    std::string solver_type = "example_solver"; // Default
+    controller_nh.getParam("ik_solver", solver_type);
```

Comment:
Absolutely minor but I personally prefer

```suggestion
    std::string solver_type =  controller_nh.param("ik_solver", "example_solver");
```

In my opinion that's cleaner to read as there is the assignment to the actual variable inside the call. Bug feel free to ignore this comment.

Reply:
I like that, but I need
```c++
    std::string solver_type = controller_nh.param("ik_solver", std::string("example_solver"));
```
for type substitution to work.

Reply:
right, strings... An alternative, I always use would be obviously
```C++
std::string solver_type =  controller_nh.param<std::string>("ik_solver", "example_solver");
```